src/paillier.rs

```diff
@@ -59,25 +63,52 @@ impl PaillierCiphertext {
 }
 
 #[derive(Clone, Debug, Serialize, Deserialize)]
-pub(crate) struct PaillierEncryptionKey(pub(crate) libpaillier::EncryptionKey);
+pub(crate) struct PaillierEncryptionKey {
+    pub(crate) pk: libpaillier::EncryptionKey,
+}
 
 impl PaillierEncryptionKey {
     pub(crate) fn n(&self) -> &BigNumber {
-        self.0.n()
+        self.pk.n()
+    }
+
+    /// Get a reference to the internal `libpaillier` encryption key.
+    pub(crate) fn as_encryption_key(&self) -> &libpaillier::EncryptionKey {
```

Comment:
#64 will get rid of the need to manipulate the `libpaillier::EncryptionKey` directly in `participant.rs`, so I don't think this code is needed (or we can just remove it when that PR gets merged in).

Reply:
Yeah, that's fair, I just saw that lovely PR that demolishes all of this. I will put it back as it was. Made this change when I tried to add `half_n` as a field on the keys instead of a method.